File: dsa/9.stack_and_queue/933_Number_of_Recent_Calls.java
package stack_and_queue;
import java.util.*;

class RecentCounter {

    List<Integer> pingList;
    int curInd;

    public RecentCounter() {
        this.pingList = new ArrayList<Integer>();
        this.curInd = 0;
    }
    
    public int ping(int t) {
        this.pingList.add(t);
        int lowerVal = t - 3000;
        while (this.pingList.get(this.curInd) < lowerVal){
            this.curInd += 1;
        }
        return this.pingList.size() - this.curInd;
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */

File: dsa/9.stack_and_queue/933_Number_of_Recent_Calls.py
from collections import deque

class RecentCounter:

    def __init__(self):
        self.queue: deque = deque()

    def ping(self, t: int) -> int:
        lower_t: int = t - 3000
        self.queue.append(t)
        while self.queue:
            if self.queue[0] < lower_t:
                self.queue.popleft()
            else:
                break
        return len(self.queue)


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)